src/pybind/mgr/mgr_util.py

```diff
@@ -1,6 +1,15 @@
 import contextlib
 import socket
 
+try:
+    from more_itertools import pairwise
+except ImportError:
+    def pairwise(iterable):
+        from itertools import tee
+        a, b = tee(iterable)
+        next(b, None)
```

Comment:
(I know this code snippet was already there, and that it's pretty straightforward, but...) this is what I meant the other day in the F2F: as we don't want to deal with the burden of packaging dependencies for all systems Ceph has to run into (because it's a real pain), we end up copy-pasting [recipes from some open source library](https://github.com/erikrose/more-itertools) and infringing its license (`The above copyright notice and this permission notice shall be included in all copies or substantial portions of the Software.`, _yeah, this maybe is just a tiny portion_) [[more-itertools LICENSE]](https://github.com/erikrose/more-itertools/blob/93e50a42b9d7ae77595918d6951cd876fee92473/LICENSE#L10-L11).

I've had to do this too for `pluggy` (as I couldn't find packages for them in Trusty/CentOS7/...), but I'm not proud of it :sweat_smile: Don't get me wrong, I really think reusing code is the right approach, but we shouldn't do it surreptitiously. IMHO we should face it openly and discuss how to do it, rather than leaving it up to every developer good will.

IMHO this is not better than distributing a dependency outside the distro package manager (e.g.: via pip). Is anyone tracking this code snippet for vulnerabilities reported in the source package? I'm afraid not. It requires manual review and update. Could we benefit from reusing other open source libraries? I think we could and should.

Reply:
We could make this a hard dependency, yes.

Reply:
Given there's no source for this package in CentOS/RHEL (AFAICT), and we haven't agreed on another solution to increase usage of existing Python libraries (in the same way we do for NPM packages), let's go with the current approach (I'd prepend the copyright notice, just to comply with MIT license).

Reply:
fyi, this `pairwise` originates from `https://docs.python.org/3/library/itertools.html#itertools-recipes` and not from more-itertools